.github/workflows/test_in_devenv.yml

```diff
@@ -0,0 +1,48 @@
+on:
+  workflow_call:
+    inputs:
+      compiler:
+        required: true
+        type: string
+
+name: Run CI within the dev environment container
+
+jobs:
+  set_up:
+    uses: ./.github/workflows/load_devdeps.yml
+    with:
+      toolchain: ${{ inputs.compiler }}
+      image_name: nvidia/cuda-quantum-devdeps
+      cache_name: ${{ github.ref_name  }}-cuda-quantum-devdeps-${{ inputs.compiler }}
+
+  run_ci:
+    runs-on: ubuntu-latest
+    needs: set_up
+    permissions:
+      contents: read
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Restore dev environment
+        uses: actions/cache/restore@v3
+        with:
+          path: /tmp/devdeps.tar
+          key: ${{ github.sha }}-devdeps-${{ inputs.compiler }}-tar
+          fail-on-cache-miss: true
+
+      - name: Create dev environment container
+        run: |
+          docker load --input /tmp/devdeps.tar
+          docker build -t cuda-quantum-devdeps -f .devcontainer/Dockerfile . --build-arg tag=${{ inputs.compiler }}-latest --build-arg workspace=. --build-arg destination=workspaces/cuda-quantum
+
+      - name: Build CUDA Quantum and run tests
```

Comment:
It would be good to separate building and running into different steps

Reply:
I split it into different steps as you suggested. Ultimately, we should also set up some tests that run in the runtime environment only, opposed to directly in the build environment, but not for this PR.

Reply:
I separated build and test into two different steps by using docker exec instead of the GH action for the build. I came across [this](https://github.com/docker/for-linux/issues/466) but the time seems comparable to before - though 12 min for the build compared to 3min for the tests...